### PR TITLE
Domain validation

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- validates the domain yaml file and raises an exception if the file has an improper format (prevents strange errors if e.g. the utterance templates are an array instead of a map)

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the changelog

### CODE REVIEW

File: rasa_core/dispatcher.py [hunk 1]
```diff
@@ -37,12 +37,25 @@ def __init__(self, sender, output_channel, domain):
...
+        # type: (Dict[Text, Any]) -> None
         """Send a message to the client."""
 
+        if message.get("buttons"):
+            self.utter_button_message(message.get("text"),
+                                      message.get("buttons"))
+        elif message.get("image"):
+            self.utter_message(message.get("text"))
+            self.utter_attachment(message.get("image"))
+        else:
```
Comment (akelad) [inline rasa_core/dispatcher.py]: what if the user specifies both buttons and an image?

Comment (tmbo) [reply rasa_core/dispatcher.py]: very good point :+1:

**Comment by tmbo** [comment]
@akelad btw. this PR includes #44 so if you reviewed this one, you can skip the other one

**Comment by znat** [comment]
This breaks something for me: 
When I run a training, `TemplateDomain` looks for`domain_schema.yml` in the current execution path, not the rasa_core path and can't find it. Any idea how to come around that?

**Comment by tmbo** [comment]
@znat that should be fixed in the latest master

**Pull request merged**